scripts/engineering/generate_phase44d_carry_aware_experimental_family_matrix_figure.py

```
@@ -0,0 +1,207 @@
+        rows = list(csv.DictReader(handle, delimiter="\t"))
+    if not rows:
+        raise SystemExit(f"no rows found in {path}")
+    return rows
+
+
+def validate_rows(rows: list[dict[str, str]], *, source: Path) -> None:
+    for row in rows:
+        if row["benchmark_version"] != EXPECTED_MATRIX_VERSION:
+            raise SystemExit(
+                f"unexpected benchmark_version in {source}: {row['benchmark_version']}"
+            )
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)